### DISCUSSION TITLE
How to set a position controller in XML file?

### DISCUSSION THREAD

**Discussion by Rukendu** [google-deepmind/mujoco#344] [Asking for Help] [answered]
Hi,

I am trying to actuate a cylinder with the help of a position controller. But it is more or less acting like a spring. Is there any way to set a fixed position of the cylinder by adjusting the gain or maybe bringing some changes in the code?

Here's the complete XML code:
[File.txt](https://github.com/deepmind/mujoco/files/8888847/File.txt)

**Comment by rohit-kumar-j** [comment]
@Rukendu you check this [playlist](https://www.youtube.com/playlist?list=PLc7bpbeTIk758Ad3fkSywdxHWpBh9PM0G)

**Reply by Rukendu** [reply]
I've checked it. It is a bit difficult to find the correct gain by keeping on  changing the values and so on. Is there any other way than randomly changing the gains?

**Comment by yuvaltassa** [comment, marked as answer]
A position controller **is** a spring.
You probably want to add `damping` to your joint.
